### PR TITLE
From Name Validation on creating e,diting and searching forms.

### PR DESCRIPTION
**Pull request by Josephalexantony-aot**
# Issue Tracking

JIRA: https://aottech.atlassian.net/browse/FWF-1972
Issue Type: BUG

# Changes

Added the validation on form name while creating a new form or editing the existing form.
Added validation on searching the form with form name consisting special charactors. 
The forms can now be created with charactors, numbers and special characters particulerly (-) and (_) only.

# How has the change been tested?

# Screenshots (if applicable)

On creating new form with special character:
image
On editing the existing form with special characters in form name will lead to validaiton error:
image
On saving the form with validation error will lead to the error toast message:
image
Form names with possible special charactors wont lead to error:
image

# Build Success screenshot (Till a CICD pipeline is set up)

# Notes

### CODE REVIEW

File: forms-flow-web/src/components/Form/Create.js [hunk 1]
```diff
@@ -221,13 +224,20 @@ const Create = React.memo(() => {
...
               <input
+                style={{ color: `${!isValidName ? "red" : ''}` }}
                 type="text"
                 className="form-control"
                 id="title"
                 placeholder={t("Enter the form title")}
                 value={form.title || ""}
-                onChange={(event) => handleChange("title", event)}
+                onChange={(event) => {
+                  setIsValidName(searchValidator(event.target.value));
```
Comment (shuhaib-aot) [inline forms-flow-web/src/components/Form/Create.js:234]: Could you please changethe name of the function searchValidator to inputValidator

**Review by shuhaib-aot — APPROVED** [review]
LGTM but please read the commnet

File: forms-flow-web/src/components/Form/Create.js (hunk 1 above)
Comment (Josephalexantony-aot) [reply forms-flow-web/src/components/Form/Create.js:234]: sure

**Review by abilpraju-aot — APPROVED** [review]

File: forms-flow-web/src/helper/regExp/inputValidater.js [hunk 2]
```diff
@@ -0,0 +1,16 @@
+const inputValidater = (searchText) => {
+    let char = '';
+    const a = searchText;
```
Comment (abhilash-aot) [inline forms-flow-web/src/helper/regExp/inputValidater.js:3]: use better naming Conventions

File: forms-flow-web/src/helper/regExp/inputValidater.js [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+    if (searchText !== '') {
+        for (var i = 0; i < a?.length; i++) {
```
Comment (abhilash-aot) [inline forms-flow-web/src/helper/regExp/inputValidater.js:5]: use let rather than var

File: forms-flow-web/src/helper/regExp/inputValidater.js [hunk 4]
```diff
@@ -0,0 +1,16 @@
...
+            char = a.charCodeAt(i);
+            if (((char > 32) && (char < 45)) || ((char > 45) && (char < 48))
```
Comment (abhilash-aot) [inline forms-flow-web/src/helper/regExp/inputValidater.js:7]: Add proper comments .   invalidChars to avoid is this right  /[^A-Za-z0-9]/g; ?

Comment (abhilash-aot) [reply forms-flow-web/src/helper/regExp/inputValidater.js:7]: Also try proper formatting

Comment (abhilash-aot) [reply forms-flow-web/src/helper/regExp/inputValidater.js:7]: `const inputValidater = (searchText) => {
  if (!searchText) {
    return false;
  }

  for (let i = 0; i < searchText.length; i++) {
    const char = searchText.charCodeAt(i);
    if (
      (char < 48) ||
      (char > 57 && char < 65) ||
      (char > 90 && char < 97) ||
      (char > 122)
    ) {
      return false;
    }
  }

  return true;
};`

**Review by abhilash-aot — CHANGES_REQUESTED** [review]
please find the comments

File: forms-flow-web/src/helper/regExp/inputValidater.js (hunk 4 above)
Comment (abhilash-aot) [reply forms-flow-web/src/helper/regExp/inputValidater.js:7]: also can update the code to a for of loop in this case , try to get the code more readable and cleaner
```
const inputValidater = (searchText) => {
  if (!searchText) {
    return false;
  }

  let valid = true;

  for (const char of searchText) {
    const charCode = char.charCodeAt(0);
    if (
      charCode < 48 ||
      (charCode > 57 && charCode < 65) ||
      (charCode > 90 && charCode < 97) ||
      charCode > 122
    ) {
      valid = false;
      break;
    }
  }

  return valid;
}
```

Comment (Josephalexantony-aot) [reply forms-flow-web/src/helper/regExp/inputValidater.js]: if (!searchText) {
    return false;
  }
adding this lines of code will return false while removing the input using backspace and will lead to the validation error like remove the special charactors in the empty input box. 
updated the code with comments and for loop.

File: forms-flow-web/src/helper/regExp/inputValidator.js [hunk 5]
```diff
@@ -0,0 +1,25 @@
+const inputValidator = (inputText) => {
+    // Set a flag to indicate whether the input is valid or not
+    let valid = true;
+    // Iterate over each character in the inputText string using a for...of loop
+    for (const char of inputText) {
```
Comment (abhilash-aot) [inline forms-flow-web/src/helper/regExp/inputValidator.js:5]: InputText is empty or null check is missing right.? Or is it handled before calling the function

**Review by abhilash-aot — CHANGES_REQUESTED** [review]
InputText empty condition need to handle.

File: forms-flow-web/src/helper/regExp/inputValidator.js (hunk 5 above)
Comment (Josephalexantony-aot) [reply forms-flow-web/src/helper/regExp/inputValidator.js:5]: it is handled on befor the function call.

**Comment by abhilash-aot** [comment]
Solve conflict

**Review by abilpraju-aot — APPROVED** [review]
LGTM

**Review by abhilash-aot — APPROVED** [review]

**Pull request merged**